### ISSUE TITLE
为什么终止任务后，仍会有进程占着显卡资源？

### ISSUE THREAD

**Issue by citisy** [Megvii-BaseDetection/YOLOX#1476]
训练了几个epoch后，使用 `ctrl + c` 终止了任务，发现部分显存仍然未释放，查看后台发现有进程残留。

**Comment by FateScript** [comment]
It' s simple if you stop your process by using `ctrl + c`. Just kill the left process with `kill` command.

**Comment by citisy** [comment]
Do you mean that I should close the program with `kill` instead of `ctrl + c` even though running the program in the foreground? I think it's too troublesome to do that every time. And then, I think I would forget to check out whether the program is still running or not sometimes.

So why the program can't stop all the process while receiving the `KeyboardInterrupt` signal?

**Comment by FateScript** [comment]
No, I mean if there exists something like GPU memory not released totally after using `ctrl + c` to stop a process, just using `kill` command to kill the process. If you find such thing everytime you are using `ctrl + c` to stop YOLOX, plz raise a issue.

As for the reason, it's often related to `nccl`.

**Comment by citisy** [comment]
Maybe it's the problem with `torch.multiprocessing`, refer to: https://pytorch.org/docs/stable/multiprocessing.html

> If the main process exits abruptly (e.g. because of an incoming signal), Python’s multiprocessing sometimes fails to clean up its children. It’s a known caveat, so if you’re seeing any resource leaks after interrupting the interpreter, it probably means that this has just happened to you.
